src/App.tsx

```diff
@@ -1,14 +1,65 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { CompletionFilter } from './types/CompletionFilter';
+
+function filterByQuery(todos: Todo[], searchQuery: string) {
+  const normalisedQuery = searchQuery.toLowerCase().trim();
+
+  return todos.filter(({ title }) => title.toLowerCase().includes(normalisedQuery));
+}
+
+function filterByCompletion(todos: Todo[], completionParam: CompletionFilter) {
+  if (completionParam === CompletionFilter.All) {
+    return todos;
+  }
+
+  const completionStatus = completionParam === CompletionFilter.Completed;
+
+  return todos.filter(({ completed }) => completed === completionStatus);
+}
+
+function postsFilter(dataToFilter: Todo[], completionParam: CompletionFilter, searchQuery: string) {
+  let filteredTodos = dataToFilter;
+
+  filteredTodos = filterByQuery(filteredTodos, searchQuery);
+  filteredTodos = filterByCompletion(filteredTodos, completionParam);
+
+  return filteredTodos;
+}
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
```

Comment:
You don't really need to make this as a state, you can use just constant for this